rust/worker/src/segment/types.rs

```diff
@@ -48,9 +381,11 @@ impl DataRecord<'_> {
     }
 }
 
-pub(crate) trait SegmentWriter {
-    fn apply_materialized_log_chunk(&self, records: Chunk<MaterializedLogRecord>);
-    fn apply_log_chunk(&self, records: Chunk<LogRecord>);
+pub(crate) trait SegmentWriter<'a> {
```

Comment:
can't let the compiler infer this lifetime?

Reply:
no this was complaining. For the distributed hnsw segment writer it was complaining with `error[E0726]: implicit elided lifetime not allowed here`. Don't understand it fully yet though